core-bundle/tests/Event/BackendSearch/IndexDocumentEventTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Event\BackendSearch;
+
+use Contao\CoreBundle\Event\BackendSearch\IndexDocumentEvent;
+use Contao\CoreBundle\Search\Backend\Document;
+use Contao\CoreBundle\Tests\TestCase;
+
+class IndexDocumentEventTest extends TestCase
+{
+    public function testCanModifyDocument(): void
+    {
+        $document = new Document('42', 'type', 'searchable');
+        $event = new IndexDocumentEvent($document);
+        $this->assertSame($document, $event->getDocument());
+        $this->assertSame([], $event->getDocument()->getMetadata());
+
+        $document = $document->withMetadata(['something-on-top-for-my-template' => 'foobar']);
```

Comment:
Now that a document/hit can be altered in an event, I wonder if the `withMetadata()` method, by default, should merge the data instead of replacing it. :thinking:

Reply:
I would prefer adding an extra method that would indicate that. Like `withMergedMetadata()` or so. For now, I wanted to KISS :)